host_core/lib/host_core/actors/actor_supervisor.ex

```diff
@@ -279,8 +280,14 @@ defmodule HostCore.Actors.ActorSupervisor do
   end
 
   # Terminate `count` instances of an actor
-  def terminate_actor(public_key, count) when count > 0 do
-    actors = Registry.lookup(Registry.ActorRegistry, public_key)
+  def terminate_actor(public_key, count, annotations) when count > 0 do
+    actors =
+      Registry.lookup(Registry.ActorRegistry, public_key)
+      |> Enum.filter(fn {pid, _v} ->
+        existing = HostCore.Actors.ActorModule.annotations(pid)
+        Map.merge(existing, annotations) == existing
```

Comment:
This might be worth a comment for anyone who's not aware of the merge case here :)